tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/core/server/OpenShiftServerUtilsTest.java

```diff
@@ -111,26 +159,54 @@ private void assertNotContainsEapLikeKeywords(String text) {
 		assertFalse(OpenShiftServerUtils.containsEapLikeKeywords(text));
 	}
 	
-	
 	@Test
 	public void testGetPodPathFromServer() {
 		// Only tests resolution from server
 		IServerAttributes server = mock(IServerAttributes.class);
 		when(server.getAttribute(OpenShiftServerUtils.ATTR_POD_PATH, (String)null)).thenReturn("test1");
 		assertEquals("test1", OpenShiftServerUtils.getPodPath(server));
 	}
+
 	@Test
 	public void testGetSourceFromServer() {
 		// Only tests resolution from server
 		IServerAttributes server = mock(IServerAttributes.class);
 		when(server.getAttribute(OpenShiftServerUtils.ATTR_SOURCE_PATH, (String)null)).thenReturn("test1");
 		assertEquals("test1", OpenShiftServerUtils.getSourcePath(server));
 	}
+
 	@Test
 	public void testGetRouteURLFromServer() {
 		// Only tests resolution from server
 		IServerAttributes server = mock(IServerAttributes.class);
 		when(server.getAttribute(OpenShiftServerUtils.ATTR_ROUTE, (String)null)).thenReturn("test1");
 		assertEquals("test1", OpenShiftServerUtils.getRouteURL(server));
 	}
+	
+	@Test
+	public void should_return_connection_from_server() {
+		// given
+		// when
+		Connection connection = OpenShiftServerUtils.getConnection(server);
+		// then
+		assertThat(connection).isEqualTo(this.connection);
+	}
+	
+	@Test
+	public void should_return_service_from_server() {
+		// given
+		// when
+		IService service = OpenShiftServerUtils.getService(server, connection);
+		// then
+		assertThat(service).isEqualTo(ResourceMocks.PROJECT2_SERVICES[1]);
+	}
+
+	@Test
+	public void should_return_deploymentconfig() throws CoreException {
+		// given
+		// when
+		IDeploymentConfig deploymentConfig = OpenShiftServerUtils.getDeploymentConfig(server);
+		// then
+		assertThat(deploymentConfig).isEqualTo(ResourceMocks.PROJECT2_DEPLOYMENTCONFIGS[2]);
+	}
```

Comment:
+1 even though the different exception cases only differ in exception msg.Trying to avoid fragile tests that assert exception message. Can assert substrings maybe to have assert exceptions and not have it too fragile